src/webgpu/api/validation/encoding/cmds/render/draw.spec.ts

```diff
@@ -348,99 +365,85 @@ success/error as expected. Such set of buffer parameters should include cases li
           },
         ],
       },
+      {
+        slot: 7,
+        stepMode: 'instance',
+        arrayStride,
+        attributes: [
+          {
+            shaderLocation: 6,
+            format: attributeFormat,
+            offset: attributeOffset,
+          },
+        ],
+      },
     ];
 
     const renderPipeline = makeTestPipeline(t, vertexBufferLayouts);
 
-    const commandBufferMaker = t.createEncoder(useBundle ? 'render bundle' : 'render pass');
-    const renderEncoder = commandBufferMaker.encoder;
-
-    if (setBufferBeforePipeline) {
-      renderEncoder.setVertexBuffer(1, vertexBuffer, setBufferOffset, setVertexBufferSize);
-    }
-    renderEncoder.setPipeline(renderPipeline);
-    if (!setBufferBeforePipeline) {
-      renderEncoder.setVertexBuffer(1, vertexBuffer, setBufferOffset, setVertexBufferSize);
-    }
+    for (const useBundle of [false, true]) {
```

Comment:
Moving the `useBundle` and `setBufferBeforePipeline` parameters into the subcase allows to reuse the resources including GPU buffers, and significantly improves the speed.